File: Telerik Algo Academy/Salaries/Salaries.js
function solve(args) {
    var N = args[0];
    var sum = 0;
    var personnel = [];
    for (var i = 0; i < N; i++) {
        personnel[i] = { str: args[1 + i], selary: 0 };
        if (personnel[i].str.indexOf('Y') == -1) {
            personnel[i].selary = 1;
            sum += personnel[i].selary;
        }
    }

    var cont = false;
    for (var i = 0; i < N; i++) {
        if (personnel[i].selary == 0) {
            var selary = 0;
            for (var j = 0; j < N; j++) {
                if (personnel[i].str[j] == 'Y') {
                    if (personnel[j].selary == 0) {
                        selary = 0;
                        break;
                    }
                    selary += personnel[j].selary;
                }
            }

            personnel[i].selary = selary;
            sum += personnel[i].selary;
            cont = true;
        }
        if (cont && i == N - 1) {
            cont = false;
            i = -1;
        }
    }

    return sum;
}

File: Telerik Algo Academy/Salaries/Salaries.cs
using System;

class Person
{
    public double Selary { get; set; }
    public string Input { get; set; }

    public Person(double selary, string input)
    {
        this.Selary = selary;
        this.Input = input;
    }
}

class Salaries
{
    static void Main()
    {
        int N = int.Parse(Console.ReadLine());
        Person[] personnel = new Person[N];
        double sum = 0;
        for (int i = 0; i < N; i++)
        {
            personnel[i] = new Person(0, Console.ReadLine());
            if (personnel[i].Input.IndexOf('Y') == -1)
            {
                personnel[i].Selary = 1;
                sum += personnel[i].Selary;
            }
        }

        bool cont = false;
        for (int i = 0; i < N; i++)
        {
            if (personnel[i].Selary == 0)
            {
                double selary = 0;
                for (int j = 0; j < N; j++)
                {
                    if (personnel[i].Input[j] == 'Y')
                    {
                        if (personnel[j].Selary == 0)
                        {
                            selary = 0;
                            break;
                        }
                        selary += personnel[j].Selary;
                    }
                }

                personnel[i].Selary = selary;
                sum += personnel[i].Selary;
                cont = true;
            }
            if (cont && i == N - 1)
            {
                cont = false;
                i = -1;
            }
        }

        Console.WriteLine(sum);
    }
}